src/main.cpp

```diff
@@ -0,0 +1,235 @@
+                co_return;
+            }
+            break;
+        case boost::beast::http::verb::delete_:
+            if (segments.empty()) {
+                std::cout << "Deletebucket detected?" << std::endl;
+            }
+            else {
+                std::cout << "Deleteobject detected" << std::endl;
+                co_await irods::s3::actions::handle_deleteobject(socket, parser, url);
+                co_return;
+            }
+            break;
+        case boost::beast::http::verb::head:
+            // Probably just headbucket and headobject here.
+            // and headbucket isn't on the immediate list
+            // of endpoints.
+            if (url.segments().empty()) {
+                std::cout << "Headbucket detected" << std::endl;
+            }
+            else {
+                std::cout << "Headobject detected" << std::endl;
+                co_await irods::s3::actions::handle_headobject(socket, parser, url);
+                co_return;
+            }
+            break;
+            /*
+        case boost::beast::http::verb::delete_:
+            // DeleteObject
+            std::cout << "Deleteobject detected" << std::endl;
+            break;
+            */
+        default:
+            std::cerr << "Oh no..." << std::endl;
+            exit(37);
+            break;
+    }
+    char buf[512];
+    while (!parser.is_done()) {
+        std::cout << "Reading" << std::endl;
+        parser.get().body().data = buf;
+        parser.get().body().size = sizeof(buf);
+        std::cout << std::string_view((char*) parser.get().body().data, parser.get().body().size) << std::endl;
+        // Using async_read here causes it to completely eat the entire input without handling it properly.
+        auto read = co_await beast::http::async_read_some(socket, buffer, parser, asio::use_awaitable);
+        std::cout << "Read " << read << " bytes" << std::endl;
+        if (ec == beast::http::error::need_buffer) {
+            ec = {};
+        }
+        else if (ec) {
+            co_return;
+        }
+    }
+    co_return;
+}
+
+unsigned short port = 8080;
+asio::awaitable<void> listener()
+{
+    auto executor = co_await this_coro::executor;
+    asio::ip::tcp::acceptor acceptor(executor, {asio::ip::tcp::v4(), port});
+    for (;;) {
+        asio::ip::tcp::socket socket = co_await acceptor.async_accept(boost::asio::use_awaitable);
+        asio::co_spawn(executor, handle_request(std::move(socket)), asio::detached);
+        std::cout << "Accepted?" << std::endl;
+    }
+}
+
+int main()
+{
+    irods::api_entry_table& api_tbl = irods::get_client_api_table();
+    irods::pack_entry_table& pk_tbl = irods::get_pack_table();
+    init_api_table(api_tbl, pk_tbl);
+
+    // This is where the configuration file is loaded and everything is set up.
+    // It should not be hard to move this to a new function wherever you like should you find that
+    // more palatable.
+    nlohmann::json config_value;
```

Comment:
Yes, let's move the loading of the configuration into a function. Makes reading `main()` easier.